### PR TITLE
Allow summary metrics to use custom time source

### PR DESCRIPTION
**Pull request by ljbade**
The `Summary` metric type uses an internal `std::chrono::steady_clock` to track when to update the metric buckets.

This PR changes it to use `system_clock` and also adds functions to allow a custom time source to be passed into the `Observe()` and `Collect()` functions.

This will allow summary metrics to be correctly calculated when replaying data.

### CODE REVIEW

**Comment by ljbade** [comment]
I still need to add test cases, but wanted to run it through CI to take a look at the coverage.

File: core/include/prometheus/histogram.h [hunk 1]
```diff
@@ -63,11 +63,20 @@ class PROMETHEUS_CPP_CORE_EXPORT Histogram {
   void ObserveMultiple(const std::vector<double>& bucket_increments,
                        const double sum_of_values);
 
-  /// \brief Get the current value of the counter.
```
Comment (ljbade) [inline core/include/prometheus/histogram.h:66]: I just made upstream PR to fix this, so will remove from this commit.

**Comment by silverjam** [comment]
@ljbade looks like this has some conflicts with master?

**Review by silverjam — APPROVED** [review]
lgtm

**Comment by ljbade** [comment]
yeah I need to rebase

**Pull request merged**